sys/include/pkt.h

```diff
@@ -0,0 +1,265 @@
+/*
+ * Copyright (C) 2014 Martin Lenders <mlenders@inf.fu-berlin.de>
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @defgroup    pkt Packet
+ * @ingroup     net
+ * @{
+ *
+ * @file    pkt.h
+ * @brief   General definitions for network packets
+ *
+ * @note    This file resides in the `sys` module's include path since it is
+ *          needed for network device drivers and the `sys/net` module's include
+ *          path is not always included (and netdev does not necessarily needs
+ *          to a compiled `pkt`).
+ *
+ * @author  Martine Lenders <mlenders@inf.fu-berlin.de>
+ */
+#ifndef __PKT_H_
+#define __PKT_H_
+
+#include <inttypes.h>
+
+#include "clist.h"
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+/**
+ * @brief   Definition of protocol families to determine the type of the packet or which protocols a network device (see @ref netdev) or protocol layer (see @ref netapi) can handle
+ *
+ * @note    XXX: The concrete definition of the values is necessary to work
+ *          with super-flexible devices as e.g. @ref native_net. It was also
+ *          decided not to use ethertype since protocols not supplied by it
+ *          might be supported
+ */
+typedef enum {
+    PKT_PROTO_UNKNOWN       = 0x0000,   /**< Type was not specified */
+
+    /**
+     * @brief   Radio frame protocol
+     *
+     * @details Sends frames as defined by radio_packet_t.
+     */
+    PKT_PROTO_RADIO         = 0x0001,
+
+    /**
+     * @brief   IEEE 802.15.4
+     *
+     * @details Sends frames as defined by ieee802154_frame_t
+     */
+    PKT_PROTO_802154        = 0x0002,
+    PKT_PROTO_6LOWPAN       = 0x0003,   /**< 6LoWPAN. */
+    PKT_PROTO_IPV6          = 0x0004,   /**< IPv6. */
+    PKT_PROTO_UDP           = 0x0005,   /**< UDP. */
+    PKT_PROTO_TCP           = 0x0006,   /**< TCP. */
+    PKT_PROTO_CCNL          = 0x0007,   /**< CCN lite. */
+
+    /**
+     * @brief   CC110x frame format protocol
+     *
+     * @details Sends frames as defined by cc110x_packet_t.
+     */
+    PKT_PROTO_CC110X        = 0x0008,
+} pkt_proto_t;
```

Comment:
How about we include some (upcoming) protocols from the beginning, i.e. IPv4, Ethernet, BLE?